lib/jerboa/client/protocol.ex

```diff
@@ -0,0 +1,265 @@
+defmodule Jerboa.Client.Protocol do
+  @moduledoc false
+
+  ## Pure functions which construct requests or indications
+  ## and processes responses
+
+  alias Jerboa.Client
+  alias Jerboa.Client.Worker
+  alias Jerboa.Client.Protocol.Transaction
+  alias Jerboa.Params
+  alias Jerboa.Format
+  alias Jerboa.Format.Body.Attribute.XORMappedAddress, as: XMA
+  alias Jerboa.Format.Body.Attribute.XORRelayedAddress, as: XRA
+  alias Jerboa.Format.Body.Attribute.{RequestedTransport, Username, Realm,
+                                      Nonce, Lifetime, ErrorCode}
+
+  ## API
+
+  @spec init_state(Client.start_opts, Worker.socket) :: Worker.state
+  def init_state(opts, socket) do
+    %Worker{
+      socket: socket,
+      server: opts[:server],
+      username: opts[:username],
+      secret: opts[:secret]
+    }
+  end
+
+  @spec bind_req(Worker.state) :: Worker.state
+  def bind_req(state) do
+    bind(state, :request)
+  end
+
+  @spec bind_ind(Worker.state) :: Worker.state
+  def bind_ind(state) do
+    bind(state, :indication)
+  end
+
+  @spec eval_bind_resp(Worker.state)
+  :: {{:ok, mapped_address :: Client.address}, Worker.state}
```

Comment:
I mean, when someone calls `Jerboa.Client.bind` it will internally call `Protocol.eval_bind_resp`, which returns a two elemenet tuple (from the Client module point of view): `{result, new_state}` - this way we just extract these two values, and return `{:reply, result, new_state}` from the `handle_call/3` callback.